### DISCUSSION TITLE
Renovate won't update python-alpine docker image when allowedVersions is set

### DISCUSSION THREAD

**Discussion by wwuck** [renovatebot/renovate#33079] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Git master e882f1fe1

### Please tell us more about your question or problem

Reproduction: https://github.com/wwuck/renovate-python-3.12.8

I have a Dockerfile for `python:3.12.7-alpine`.

```Dockerfile
FROM python:3.12.7-alpine@sha256:5049c050bdc68575a10bcb1885baa0689b6c15152d8a56a7e399fb49f783bf98
```

I have added a package rule to prevent python updates to `3.13`.

```json
    "packageRules": [
        {
            "allowedVersions": "<3.13",
            "description": "Block python 3.13",
            "matchPackageNames": [
                "python"
            ]
        }
    ]
```

I am expecting renovate to update the python docker image from `3.12.7-alpine` to `3.12.8-alpine`.

When I run renovate on the reproduction, it doesn't create any update branch or PR.

### Logs (if relevant)

<details><summary>Logs</summary>

```
DEBUG: packageFiles with updates (repository=wwuck/renovate-python-3.12.8, baseBranch=main)
       "config": {
         "dockerfile": [
           {
             "deps": [
               {
                 "depName": "python",
                 "currentValue": "3.12.7-alpine",
                 "currentDigest": "sha256:5049c050bdc68575a10bcb1885baa0689b6c15152d8a56a7e399fb49f783bf98",
                 "replaceString": "python:3.12.7-alpine@sha256:5049c050bdc68575a10bcb1885baa0689b6c15152d8a56a7e399fb49f783bf98",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "final",
                 "updates": [],
                 "packageName": "python",
                 "versioning": "docker",
                 "warnings": [],
                 "registryUrl": "https://index.docker.io",
                 "lookupName": "library/python",
                 "currentVersion": "3.12.7",
                 "currentVersionTimestamp": "2024-11-13T07:09:51.144Z",
                 "currentVersionAgeInDays": 34,
                 "fixedVersion": "3.12.7-alpine"
               }
             ],
             "packageFile": "Dockerfile"
           }
         ]
       }
```

</details>

**Comment by rarkins** [comment]
Does the problem go away if the package rule with allowedVersions is removed?

**Reply by wwuck** [reply]
I removed the package rule on the reproduction repo and I can see renovate has changed it to a PR for `3.13.1` on all python image versions. The `3.12.7-alpine` and `3.12.7-slim` are now in the PR updated to `3.13.1`.

Renovate also still keeps the additional PR for updating the digest on `python:3.12.7` and `python:3.12.7-slim` image versions. I was expecting renovate to ignore the digest update and only show the `3.13.1` update but maybe I am not understanding how renovate works correctly in that case.

**Reply by rarkins** [reply]
I'm not sure what the problem is or if there's still a problem, but the above is an example of why you should always make _minimal_ reproductions. You had non-minimal config, which directly affected the behavior, and you also left it out of your description.

**Reply by wwuck** [reply]
Yeah sorry, I've done a new minimal reproduction https://github.com/wwuck/renovate-python-3.12.8.

The Dockerfile image is `3.12.7-alpine` and renovate does not update this to `3.12.8-alpine`.

**Reply by rarkins** [reply]
Please update the title and description of this discussion to match.

**Reply by wwuck** [reply]
I removed the slim/multiarch dockerfile for this reproduction.

**Reply by rarkins** [reply, marked as answer]
I've found a way that works: https://github.com/renovate-reproductions/33079/blob/b5f1a59e099043fa66ca845d1e8dc42a333e95ad/renovate.json#L6

**Reply by wwuck** [reply]
I can verify that appears to work here too.
